src/content/en/2021/caching.md

```diff
@@ -17,4 +19,250 @@ featured_stat_3: TODO
+  height=501,
+  sheets_gid="58739923",
+  sql_file="top_cdns.sql"
+) }}
+
+## Service Worker adoption
+
+A service worker is a script that the browser allows to run in the background independently from a web page. It supports features that don't need direct user or web page interaction and offer rich offline experiences.
+
+The adoption of Service Workers has continued to steadily increase. While 1% of Mobile pages registered a service worker in 2020, 9% of Mobile pages ranked in the top 1,000 registered one in 2021.
+
+{{ figure_markup(
+  image="sw-adoption.png",
+  caption="Service Worker adoption.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=688062455&format=interactive",
+  sheets_gid="802609299",
+  sql_file="service_worker_rank.sql"
+) }}
+
+The primary way to cache resources within a Service Worker, it's by using the CacheStorage API. This allows a developer to create a custom cache strategy for any requests passing through the worker; some famous ones are Stale-While-Revalidate, Cache Falling Back to Network, Network Falling Back to Cache, Cache Only. In recent years become even easier to adopt those strategies by increasing the popularity of [Workbox](https://developers.google.com/web/tools/workbox/modules/workbox-strategies), which decides what cache you want to plug and play.
+
+## Headers adoption
+
+With both CDN and Browser, the HTTP Headers are the leading toolkit each developer needs to master to properly cache resources. Headers are simply instructions known as "directives", read during the HTTP Request or Response to control the cache strategy.
+
+The caching-related headers, or the absence of them, tell the browser or CDN three essential pieces of information:
+- **Cacheability**: Is this content cacheable?
+- **Freshness**: If it is cacheable, how long can it be cached for? `Cache-Control` and `Expires` 
+- **Validation**: If it is cacheable, how do I ensure that my cached version is still fresh?
+
+Headers are meant to be used either alone or together. To tell if the content is cachable and fresh, we have:
+- `Expires` specifies an explicit expiration date and time (i.e., when precisely the content expires)
+- `Cache-Control` specifies a cache duration (i.e., how long the content can be cached in the browser relative to when it was requested)
+In case both are specified, `Cache-Control` takes precedence.
+
+The usage of the Cache-Control header has increased steadily since 2019. 74% of responses on Mobile pages included the Cache-Control header, while 75% of responses on Desktop pages utilized the header. From 2020, the usage of this specific header increased by 0.71% for Mobile and by 1.13% for Desktop.
+
+Cache-Control: DT 75%, Mobile 74%
```

Comment:
@rviscomi two main figure I think they are going to be useful to have is showing:
Figure 1: Usage of Cache-Control and Expires headers similar to https://almanac.httparchive.org/static/images/2020/caching/cache-control-and-max-age-and-expires.png

Figure 2: Usage of Last-Modified and ETag headers similar to https://almanac.httparchive.org/static/images/2020/caching/last-modified-and-etag.png


I believe the data is here https://docs.google.com/spreadsheets/d/1-v3yR0LZIC3t4zWtqTgR3jJsKjjRMP-HATU2caP8e2c/edit#gid=2102749619

Also I live it to you decide we want to have in the chart something like desktop vs mobile or 2020 vs 2021.

Reply:
Added. I think adding another time dimension to the charts might make them hard to read, so just sticking with 2021 for simplicity. If you wanted to refer back to 2020 data you could link to the corresponding section from that year's chapter.

Reply:
Perfect, thank you!